hal/architecture/MyHwNRF5.cpp

```diff
@@ -204,16 +207,25 @@ void hwSleepPrepare(unsigned long ms)
 		// Reset RTC
 		MY_HW_RTC->TASKS_CLEAR = 1;
 
-		// Calculate sleep time
-		// 8 Hz -> max 582.542 hours sleep.
-		MY_HW_RTC->PRESCALER = 4095;
-		// Set compare register to 1/125ms + 2 to garantee event triggering
-		MY_HW_RTC->CC[0] = (ms / 125) + 2;
+		// Calculate sleep time and prescaler
+		if (ms<512000) {
+			// prescaler 0, 30.517 μs resolution -> max 512 s sleep
+			MY_HW_RTC->PRESCALER =  0;
+			// Set compare register to 1/30.517 µs to garantee event triggering
+			// A minimum of 2 ticks must be guaranteed
+			// (1000/32768)<<12 == 125
+			MY_HW_RTC->CC[0] = min((ms<<12 / 125), 2);
+		} else {
+			// 8 Hz -> max 582.542 hours sleep.
+			MY_HW_RTC->PRESCALER = 4095;
+			// Set compare register to 1/125ms
+			// A minimum of 2 ticks must be guaranteed
+			MY_HW_RTC->CC[0] = min((ms / 125), 2);
```

Comment:
The comment says "A minimum of 2 ticks must be guaranteed" but min(a,b) will return the smaller value. Should this line be
MY_HW_RTC->CC[0] = max((ms / 125), 2);
instead?

Reply:
Thank you. I fix this.